actions/shared_publish_to_docker/action.yaml

```diff
@@ -0,0 +1,117 @@
+name: Shared publish to Docker
+description: Shared action for publish image to Docker
+
+inputs:
+  new_version:
+    description: The new version number to publish for the docker repo
+    required: true
+  image_tag:
+    description: The image tag used to extract metadata for Docker
+    required: true
+  not_snapshot:
+    description: True if the release type is not a snapshot or pre-release build
+    default: 'false'
+  github_token:
+    description: The github token to login to Docker container
+    required: true
+  docker_file:
+    description: The Dockerfile used to build and publish Docker image
+    default: ''
+  publish_vulnerabilities:
+    description: If true, will attempt to publish any vulnerabilities to GitHub. Defaults to true. Set to false for private repos.
+    type: string
+    default: true
+  docker_image_name:
+    description: The docker image name
+    required: true
+    type: string
+  docker_registry:
+    description: The docker registry
+    required: true
+    type: string
+
+outputs:
+  tags:
+    description: The docker image tags
+    value: ${{ steps.meta.outputs.tags }}
+
+runs:
+  using: "composite" 
+
+  steps:
+    - name: Log in to the Docker container registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ inputs.docker_registry }}
+        username: ${{ github.actor }}
+        password: ${{ inputs.github_token }}
+
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ inputs.docker_registry }}/${{ inputs.docker_image_name }}
+        tags: |
+          type=raw,value=${{ inputs.image_tag }}
+          type=raw,value=${{ inputs.not_snapshot == 'true' && 'latest' || '' }}
+
+    - name: Extract First Image Tag
+      id: extractImageTag
+      shell: bash
+      run: |
+        FIRST_TAG=${DOCKER_METADATA_OUTPUT_TAGS%%$'\n'*}
+        echo "firstTag=$FIRST_TAG" >> $GITHUB_OUTPUT
+
+    - name: Build and export to Docker
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: ${{ inputs.docker_file }}
+        load: true
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+        build-args: |
+          JAR_VERSION=${{ inputs.new_version }}
+          IMAGE_VERSION=${{ inputs.new_version }}
+
+    ### These steps should use the vulnerability_scan_filesystem action.
+    ### Refactor this as a part of UID2-2681.
+    # - name: Generate Trivy vulnerability scan report
```

Comment:
Disabling these because vulnerability is found in tcp-portal repo: https://github.com/IABTechLab/uid2-tcportal/actions/runs/7565806804/job/20602130093

Reply:
It would be better to make the severity for the failure to be an input, and then in TC Portal we can set it to only fail on critical. We will most likely want to have different levels for different repos anyway